File: airesources/C++/MyBot.cpp
#include <stdlib.h>
#include <time.h>
#include <cstdlib>
#include <ctime>
#include <time.h>
#include <set>
#include <fstream>

#include "hlt.hpp"
#include "networking.hpp"

int main(int argc, char ** argv) {
	srand(time(NULL));

	std::ofstream dbg("dbg.log");

	std::cout.sync_with_stdio(0);

	unsigned char myID;
	hlt::GameMap presentMap;
	getInit(atoi(argv[1]), myID, presentMap); //argv[1] is the port number to connect to.
	sendInit("C++Bot");

	std::set<hlt::Move> moves;
	while(true) {
		moves.clear();

		getFrame(presentMap);

		for(unsigned short a = 0; a < presentMap.height; a++) {
			for(unsigned short b = 0; b < presentMap.width; b++) {
				if (presentMap.getSite({ b, a }).owner == myID) {
					moves.insert({ { b, a }, (unsigned char)(rand() % 5) });
				}
			}
		}

		sendFrame(moves);
	}

	return 0;
}

File: airesources/Python/MyBot.py
from hlt import *
from networking import *
import sys

myID, gameMap = getInit(int(sys.argv[1])) # Port is first argument.
sendInit("PythonBot")

while True:
	moves = []
	gameMap = getFrame()
	for y in range(gameMap.height):
		for x in range(gameMap.width):
			if gameMap.getSite(Location(x, y)).owner == myID:
				moves.append(Move(Location(x, y), int(random.random() * 5)))
	sendFrame(moves)
